tests/SpecTests/FunctionalTestCase.php

```diff
@@ -137,8 +141,9 @@ protected function checkServerRequirements(array $runOn): void
             $minServerVersion = $req->minServerVersion ?? null;
             $maxServerVersion = $req->maxServerVersion ?? null;
             $topologies = $req->topology ?? null;
+            $serverlessMode = $req->serverless ?? null;
 
-            if ($this->isServerRequirementSatisifed($minServerVersion, $maxServerVersion, $topologies)) {
+            if ($this->isServerRequirementSatisifed($minServerVersion, $maxServerVersion, $topologies, $serverlessMode)) {
```

Comment:
Should serverless pre-empt any requirement checking for server version? Feel free to point me to some language in the spec test READMEs that say as much. I don't recall the requirements being mutually exclusive, though.

Reply:
Theoretically, the check should pre-empt any other checks, as it's entirely based on the environment variable being set or not. That said, the unified test format currently prescribes that the requirements MUST be checked in the following order:
1. minServerVersion
2. maxServerVersion
3. topologies
4. serverless
5. serverParameters
6. auth

I don't think this order is efficient or even necessary, so I'm happy to have the spec updated to suggest a more efficient order without mandating it (to avoid unnecessary downstream changes in drivers)

Reply:
Reported https://jira.mongodb.org/browse/DRIVERS-1885

Reply:
I'd defer that work to a separate PR in an interest to get this merged.